Terminal.Gui/Drawing/Thickness.cs

```diff
@@ -1,298 +1,273 @@
-		/// </summary>
-		/// <param name="other"></param>
-		/// <returns>true if the current object is equal to the other parameter; otherwise, false.</returns>
-		public bool Equals (Thickness other)
-		{
-			return other is not null &&
-			       Left == other.Left &&
-			       Right == other.Right &&
-			       Top == other.Top &&
-			       Bottom == other.Bottom;
-		}
-
-		/// <inheritdoc/>
-		public override int GetHashCode ()
-		{
-			int hashCode = 1380952125;
-			hashCode = hashCode * -1521134295 + Left.GetHashCode ();
-			hashCode = hashCode * -1521134295 + Right.GetHashCode ();
-			hashCode = hashCode * -1521134295 + Top.GetHashCode ();
-			hashCode = hashCode * -1521134295 + Bottom.GetHashCode ();
-			return hashCode;
-		}
-
-		/// <inheritdoc/>
-		public static bool operator == (Thickness left, Thickness right)
-		{
-			return EqualityComparer<Thickness>.Default.Equals (left, right);
-		}
-
-		/// <inheritdoc/>
-		public static bool operator != (Thickness left, Thickness right)
-		{
-			return !(left == right);
-		}
-	}
-}
\ No newline at end of file
+﻿using System.Text.Json.Serialization;
+
+namespace Terminal.Gui;
+
+/// <summary>
+///     Describes the thickness of a frame around a rectangle. Four <see cref="int"/> values describe the
+///     <see cref="Left"/>, <see cref="Top"/>, <see cref="Right"/>, and <see cref="Bottom"/> sides of the rectangle,
+///     respectively.
+/// </summary>
+/// <remarks>
+///     <para>
+///         Use the helper API (<see cref="GetInside(Rect)"/> to get the rectangle describing the insides of the frame,
+///         with the thickness widths subtracted.
+///     </para>
+///     <para>Use the helper API (<see cref="Draw(Rect, string)"/> to draw the frame with the specified thickness.</para>
+/// </remarks>
+public class Thickness : IEquatable<Thickness> {
+    /// <summary>Initializes a new instance of the <see cref="Thickness"/> class with all widths set to 0.</summary>
+    public Thickness () { }
+
+    /// <summary>Initializes a new instance of the <see cref="Thickness"/> class with a uniform width to each side.</summary>
+    /// <param name="width"></param>
+    public Thickness (int width) : this (width, width, width, width) { }
+
+    /// <summary>
+    ///     Initializes a new instance of the <see cref="Thickness"/> class that has specific widths applied to each side of
+    ///     the rectangle.
+    /// </summary>
+    /// <param name="left"></param>
+    /// <param name="top"></param>
+    /// <param name="right"></param>
+    /// <param name="bottom"></param>
+    public Thickness (int left, int top, int right, int bottom) {
+        Left = left;
+        Top = top;
+        Right = right;
+        Bottom = bottom;
+    }
+
+    /// <summary>Gets or sets the width of the lower side of the rectangle.</summary>
+    [JsonInclude]
+    public int Bottom;
+
+    /// <summary>Gets or sets the width of the left side of the rectangle.</summary>
+    [JsonInclude]
```

Comment:
(out of scope)

Fields on reference types typically shouldn't be public.

Either this type should be a `struct`/`record struct` (likely) or these should be exposed as properties for the public API surface.